### PR TITLE
Fix sysconfig not being handled correctly by sysvinit

### LINKED ISSUE #6362: safe-reload does not know about env-vars in a sysvinit environment
Had to put the following to the exec command in the sysvinit-script (reload function) to get safe-reload to work:

`env ICINGA2_PID_FILE=$ICINGA2_PID_FILE DAEMON=$DAEMON`

```sh
reload() {
    exec env ICINGA2_PID_FILE=$ICINGA2_PID_FILE DAEMON=$DAEMON @CMAKE_INSTALL_PREFIX@/lib/icinga2/safe-reload
}
```

This is with the version from the current master.

### PR DESCRIPTION
**Pull request by Crunsher**
TODO: test

fixes #6362

### CODE REVIEW

File: etc/initsystem/prepare-dirs.cmake [hunk 1]
```diff
@@ -3,6 +3,11 @@
 # This script prepares directories and files needed for running Icinga2
 #
 
+# With sysvinit we need to pass the environment this way. Or do we?
```
Comment (dnsmichi) [inline etc/initsystem/prepare-dirs.cmake]: If this gets merged, the comment should be updated.

File: etc/initsystem/prepare-dirs.cmake [hunk 2]
```diff
@@ -3,6 +3,11 @@
...
+if [ "$1" != "" ]; then
+	source "$1"
+fi
+
```
Comment (dnsmichi) [inline etc/initsystem/prepare-dirs.cmake]: This does only check whether a parameter was given. v2.8.4 checks if the file actually exists.

https://github.com/Icinga/icinga2/blob/78322b095310bb95c2015ea03457c52d327991e8/etc/initsystem/prepare-dirs#L6

**Comment by dnsmichi** [comment]
@Crunsher please discuss this with @lazyfrosch today.

File: etc/initsystem/prepare-dirs.cmake (hunk 2 above)
Comment (Crunsher) [reply etc/initsystem/prepare-dirs.cmake]: Good point

File: etc/initsystem/prepare-dirs.cmake (hunk 1 above)
Comment (Crunsher) [reply etc/initsystem/prepare-dirs.cmake]: Aye, the ramblings of my thought process are not really required here ^_^

**Comment by Crunsher** [comment]
A note, I have recently read using `#!/bin/sh` should be avoided, instead the path ought to be read from the environment for better portability.

**Comment by dnsmichi** [comment]
The discussion env vs path should be out of the scope of this issue to be honest. I'm not keen on breaking things with 2.9.

**Pull request merged**